packages/vite/src/node/importGlob.ts

```diff
@@ -261,3 +257,36 @@ function error(pos: number) {
   err.pos = pos
   throw err
 }
+
+function glob(pattern: string, cwd: string): string[] {
+  const rebase = rebaseGlobPattern(pattern)
+  if (rebase !== '') {
+    pattern = pattern.slice(rebase.length)
+    cwd = path.posix.join(cwd, rebase)
+  }
+
+  let files = fastGlob.sync(pattern, {
+    cwd,
+    ignore: ['**/node_modules/**']
+  })
+
+  if (rebase !== '') {
+    files = files.map((file) => path.posix.join(rebase, file))
+  }
+
+  return files
+}
+
+// Make `ignore: ['**/node_modules/**']` work with patterns that include `node_modules/`.
+// E.g. `import.meta.glob('node_modules/framework/**/*.page.js')`.
+function rebaseGlobPattern(pattern: string) {
+  let rebase = ''
+  if (pattern.includes('/node_modules/')) {
+    rebase =
+      pattern.split('/node_modules/').slice(0, -1).join('/node_modules/') +
+      '/node_modules/'
+  } else if (pattern.startsWith('node_modules/')) {
+    rebase = 'node_modules/'
+  }
+  return rebase
+}
```

Comment:
Maybe we could use a regex here?

```suggestion
const nodeModulesRE = /.*\/?node_modules\//
function rebaseGlobPattern(pattern: string) {
  return pattern.match(nodeModulesRE)?.[0] ?? ''
}
```

Reply:
Done. I also squashed `rebaseGlobPattern()` with `glob().